pywemo/subscribe.py

```diff
@@ -188,12 +188,12 @@ def _send_response(self, code, body, *, content_type="text/html"):
             self.wfile.write(body.encode("UTF-8"))
 
     def _get_xml_from_http_body(self):
-        """Build the cElementTree root from the body of the http request."""
+        """Build the element tree root from the body of the http request."""
         content_len = int(self.headers.get('content-length', 0))
         data = self.rfile.read(content_len)
         # trim garbage from end, if any
-        data = data.decode("UTF-8").strip()
-        return cElementTree.fromstring(data)
+        data = data.decode('UTF-8').strip().encode('UTF-8')
```

Comment:
data = data.strip()

I think the decode/encode is unnecessary.

Reply:
Looking at this again, the `.strip()` might be unnecessary too. Maybe lxml is more forgiving about trailing/leading whitespace?

```
>>> from lxml import etree as et
>>> et.fromstring(b'\r\n\t  <a><b>c</b></a>   \r\n\t').tag                                                                                            
'a'
```

Reply:
Never mind. The `.strip()` is still needed.  Here's another example.

```
from lxml import etree as et
>>> et.fromstring(b'\r\n\t  <?xml version="1.0" encoding="utf-8"?><a><b>c</b></a>   \r\n\t').tag
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "src/lxml/etree.pyx", line 3237, in lxml.etree.fromstring
  File "src/lxml/parser.pxi", line 1896, in lxml.etree._parseMemoryDocument
  File "src/lxml/parser.pxi", line 1784, in lxml.etree._parseDoc
  File "src/lxml/parser.pxi", line 1141, in lxml.etree._BaseParser._parseDoc
  File "src/lxml/parser.pxi", line 615, in lxml.etree._ParserContext._handleParseResultDoc
  File "src/lxml/parser.pxi", line 725, in lxml.etree._handleParseResult
  File "src/lxml/parser.pxi", line 654, in lxml.etree._raiseParseError
  File "<string>", line 2
lxml.etree.XMLSyntaxError: XML declaration allowed only at the start of the document, line 2, column 9
```

The `<?xml version="1.0" encoding="utf-8"?>` is what is causing problems for the Bridge device. The result from `end_devices.get('DeviceLists')` includes that encoding declaration at the start. I think the uses of `et.fromstring` in humidifier.py, coffeemaker.py & maker.py are all safe, as they don't include this encoding declaration.

Reply:
Ya, I ran into the encoding be defined in the XML issue as well at one point and thought I fixed it, but I figured there was some issues that probably slipped through.

Reply:
ok, I removed the decode/encode and just strip it directly now.